crates/autopilot/src/infra/persistence/mod.rs

```diff
@@ -18,4 +24,83 @@ impl Persistence {
             postgres,
         }
     }
+
+    /// There is always only one `current` auction.
+    ///
+    /// This method replaces the current auction with the given one.
+    ///
+    /// If the given auction is successfully saved, it is also archived.
+    pub async fn replace_current_auction(
+        &self,
+        auction: domain::Auction,
+    ) -> Result<domain::AuctionId, Error> {
+        self.postgres
+            .replace_current_auction(&dto::auction::from_domain(auction.clone()))
+            .await
+            .map(|auction_id| {
+                self.archive_auction(auction_id, auction);
+                auction_id
+            })
+            .map_err(Error::DbError)
+    }
+
+    /// Saves the given auction to storage for debugging purposes.
+    ///
+    /// There is no intention to retrieve this data programmatically.
+    fn archive_auction(&self, id: domain::AuctionId, instance: domain::Auction) {
+        if let Some(uploader) = self.s3.clone() {
```

Comment:
nit: This would remove 1 level of indentation.
```suggestion
        let Some(uploader) = self.s3.clone() else {
            return;
        };
```